tests/modules/images/regenerate-existing-images/background-process/perflab-background-job-test.php

```diff
@@ -0,0 +1,228 @@
+<?php
+/**
+ * Tests for background-job module.
+ *
+ * @package performance-lab
+ * @group   background-process
+ */
+
+/**
+ * Class Perflab_Background_Job_Test
+ *
+ * @coversDefaultClass Perflab_Background_Job
+ * @group regenerate-existing-images
+ */
+class Perflab_Background_Job_Test extends WP_UnitTestCase {
+	/**
+	 * Job instance.
+	 *
+	 * @var Perflab_Background_Job
+	 */
+	private $job;
+
+	/**
+	 * Runs before any test is executed inside class.
+	 *
+	 * @return void
+	 */
+	public static function set_up_before_class() {
+		require_once PERFLAB_PLUGIN_DIR_PATH . 'modules/images/regenerate-existing-images/background-process/class-perflab-background-job.php';
+	}
+
+	public function test_class_constants_exists() {
+		$job_class = Perflab_Background_Job::class;
+
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_NAME' ) );
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_DATA' ) );
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_ATTEMPTS' ) );
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_LOCK' ) );
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_ERRORS' ) );
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_STATUS' ) );
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_COMPLETED_AT' ) );
+		$this->assertTrue( defined( $job_class . '::JOB_STATUS_QUEUED' ) );
+		$this->assertTrue( defined( $job_class . '::JOB_STATUS_RUNNING' ) );
+		$this->assertTrue( defined( $job_class . '::JOB_STATUS_PARTIAL' ) );
+		$this->assertTrue( defined( $job_class . '::JOB_STATUS_COMPLETE' ) );
+		$this->assertTrue( defined( $job_class . '::JOB_STATUS_FAILED' ) );
+	}
+
+	public function test_class_constants_values() {
+		$job_class = Perflab_Background_Job::class;
+
+		$this->assertEquals( 'perflab_job_name', constant( $job_class . '::META_KEY_JOB_NAME' ) );
+		$this->assertEquals( 'perflab_job_data', constant( $job_class . '::META_KEY_JOB_DATA' ) );
+		$this->assertEquals( 'perflab_job_attempts', constant( $job_class . '::META_KEY_JOB_ATTEMPTS' ) );
+		$this->assertEquals( 'perflab_job_lock', constant( $job_class . '::META_KEY_JOB_LOCK' ) );
+		$this->assertEquals( 'perflab_job_errors', constant( $job_class . '::META_KEY_JOB_ERRORS' ) );
+		$this->assertEquals( 'perflab_job_status', constant( $job_class . '::META_KEY_JOB_STATUS' ) );
+		$this->assertEquals( 'perflab_job_completed_at', constant( $job_class . '::META_KEY_JOB_COMPLETED_AT' ) );
+		$this->assertEquals( 'perflab_job_queued', constant( $job_class . '::JOB_STATUS_QUEUED' ) );
+		$this->assertEquals( 'perflab_job_running', constant( $job_class . '::JOB_STATUS_RUNNING' ) );
+		$this->assertEquals( 'perflab_job_partial', constant( $job_class . '::JOB_STATUS_PARTIAL' ) );
+		$this->assertEquals( 'perflab_job_complete', constant( $job_class . '::JOB_STATUS_COMPLETE' ) );
+		$this->assertEquals( 'perflab_job_failed', constant( $job_class . '::JOB_STATUS_FAILED' ) );
+	}
+
+	/**
+	 * @covers ::set_status
+	 */
```

Comment:
Please remove all the `@covers` annotations throughout the class, as they are not used in WordPress.

Reply:
Per [56070](https://core.trac.wordpress.org/ticket/56070) we can use `@covers` in unit test. I got review for this in one of my PR for WordPress core https://github.com/WordPress/wordpress-develop/pull/3034#discussion_r965434551

@felixarntz Please correct me if I'm wrong.

Reply:
@ankitrox @mukeshpanchal27 Fair point. However, in that case I'm not sure the declaration here is valid? The `::function` syntax works if it is for a function, but we're testing a class method here. Don't we have to therefore include the class name as well to make this work?

Reply:
@felixarntz We have `@coversDefaultClass Perflab_Background_Job` at the top of class declaration. Phpunit coverage report can understand that we are testing interface/methods of this class.

Reply:
@ankitrox Ahh - I wasn't aware of that. Thank you for clarifying, all good then :)